gtd_generative_proof_machine/inc/proof_tree.h

```diff
@@ -0,0 +1,24 @@
+#include <string.h>
+#include "common.h"
+
+typedef struct ProofNode ProofNode;
+
+typedef struct ProofTree
+{
+    int depth;
+    ProofNode *head;
+    ProofNode *tail;
+} ProofTree;
+
+typedef struct ProofNode
+{
+    const char *message;
+    ProofTree *subtree;
+    ProofNode *next;
+} ProofNode;
+
+ProofNode *create_proof_node(void);
+ProofTree *create_proof_tree(void);
+void append_proof_node(ProofTree *tree, ProofNode *node);
+void write_proof_node(ProofNode *node, FILE *outFile, int depth);
```

Comment:
Will this function be called by anyone else than write_proof_tree? If not, remove the declaration from header and make it static

Reply:
yes, all of these functions are used outside of proof_tree.c (eg. in generative_restriction.c and generative_proof_machine.c)